### PR TITLE
fix: Add functionality for new Vercel webhook payloads

### PR DESCRIPTION
**Pull request by jangjodi**
Fixes WOR-2493

Before: Only old Vercel webhook payloads were supported. This resulted in failures when events with the new Vercel webhook payloads were sent. See [WOR-2493](https://getsentry.atlassian.net/browse/WOR-2493) for more details.

Now: Code supports new and old Vercel webhook payloads

Successful `deployment.created` POST with new payload
Screen Shot 2022-12-12 at 9 56 24 AM copy

Successful `integration-configuration.removed` POST with new payload
Screen Shot 2022-12-14 at 9 25 24 AM

### CODE REVIEW

File: fixtures/vercel.py
Comment (jangjodi) [inline fixtures/vercel.py:48]: Create a mock deployment webhook response with the new payload

File: src/sentry/integrations/vercel/webhook.py
Comment (jangjodi) [inline src/sentry/integrations/vercel/webhook.py:177]: Try the Vercel request with the new event type and payload first. If it fails, try the old one. Set a tag depending on which type it is.

All functionality remains the same, just the event type and how the payload is accessed is changed

File: src/sentry/integrations/vercel/webhook.py
Comment (jangjodi) [inline src/sentry/integrations/vercel/webhook.py:192]: Same as previous comment

File: src/sentry/integrations/vercel/webhook.py
Comment (jangjodi) [inline src/sentry/integrations/vercel/webhook.py:299]: Change the way the project id is accessed so that it supports new and old payloads

File: tests/sentry/integrations/vercel/test_uninstall.py
Comment (jangjodi) [inline tests/sentry/integrations/vercel/test_uninstall.py:43]: Create a mock remove installation response with the new payload

File: tests/sentry/integrations/vercel/test_uninstall.py
Comment (jangjodi) [inline tests/sentry/integrations/vercel/test_uninstall.py:105]: Test that removing an installation works with the new payload and event type

File: tests/sentry/integrations/vercel/test_webhook.py
Comment (jangjodi) [inline tests/sentry/integrations/vercel/test_webhook.py:139]: Test that creating a deployment works with the new payload and event type

File: src/sentry/integrations/vercel/webhook.py
Comment (leeandher) [inline src/sentry/integrations/vercel/webhook.py:149]: Great call on using the sdk here!

File: src/sentry/integrations/vercel/webhook.py
Comment (leeandher) [inline src/sentry/integrations/vercel/webhook.py:164]: nit: You already have `payload` in scope, so you can just do `payload["configuration"]["id"]`

File: src/sentry/integrations/vercel/webhook.py
Comment (leeandher) [inline src/sentry/integrations/vercel/webhook.py:168]: This works checking for the KeyError but with `.get` accessors we may end up with `None` for the ext_id or conf_id and fail with some other error in `self._delete` or `self._deployment_created`.

It may be worth refactoring to ensure no KeyErrors with safe `.get`'s, and just check if the id is none before trying the legacy pattern

Some pseudocode:

```
external_id = ... #some safe .get() on new payload
if external_id is None:
   external_id = # legacy payload
   
conf_id = ... #some safe .get() on new payload
if conf_id is None:
   conf_id = # legacy payload
   
if event_type in ['integration-configuration.removed', "integration-configuration-removed"]
  # remove
if event_type in ['deployment', 'deployment.created']
  # create deploy
```

Comment (leeandher) [reply src/sentry/integrations/vercel/webhook.py:168]: This is just a suggestion, if you want to stick with the try/except approach that's totally fine as well, but you need to ensure a KeyError, or use a more generic Exception.

File: src/sentry/integrations/vercel/webhook.py
Comment (leeandher) [inline src/sentry/integrations/vercel/webhook.py:189]: Same as above!

**Review by leeandher — COMMENTED** [review]
Looks good! Few comments though, so I'll just wait for you responses and can approve after that 👍

File: src/sentry/integrations/vercel/webhook.py
Comment (jangjodi) [reply src/sentry/integrations/vercel/webhook.py]: Thanks for bringing this up!

Currently, payload is `{"team": null}` when no team exists, so `payload.get("team")` is None and the original line works. Unfortunately, I believe your suggested line did not work as a result of this.

However, I modified the line to be `external_id = (
                    payload.get("team")["id"]
                    if (payload.get("team") and payload.get("team") != {})
                    else payload["user"]["id"]
                )` in case `{"team": {}}` occurs. 
Do you think this would work?

File: src/sentry/integrations/vercel/webhook.py
Comment (jangjodi) [reply src/sentry/integrations/vercel/webhook.py]: Good catch, thank you!

File: src/sentry/integrations/vercel/webhook.py
Comment (jangjodi) [reply src/sentry/integrations/vercel/webhook.py]: Thanks for the suggestion!

I decided to keep the try/except just so the code pertaining to the old payload can be removed more easily in the future, but I implemented your suggested to make it a more generic Exception :)

**Review by leeandher — APPROVED** [review]
looks good to me! 🚀

**Pull request merged**
